### PR TITLE
Feat: Gradle plugin v2

### PR DESCRIPTION
**Pull request by marandaneto**

### CODE REVIEW

**Comment by marandaneto** [comment]
@cortinico looks like https://github.com/getsentry/sentry-android-gradle-plugin/blob/feat/gradle-plugin-v2/plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryPluginTest.kt#L103 is failing for windows, we don't need to get blocked by this, just be aware :)

**Comment by marandaneto** [comment]
@cortinico CI is not happy because of 

> Cannot set the value of extension 'sentry' property 'autoUpload' of type java.lang.Boolean using an instance of type java.lang.String.

File: examples/android-gradle/build.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        release {
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile(
+              'proguard-android-optimize.txt')
+        }
+    }
+}
+
+sentry {
+    autoUpload = System.getenv("AUTO_UPLOAD") ?: false
```
Comment (marandaneto) [inline examples/android-gradle/build.gradle:35]: needs to convert to a boolean here probably

Comment (marandaneto) [inline examples/android-gradle/build.gradle:35]:
```suggestion
    autoUpload = System.getenv("AUTO_UPLOAD").toBoolean() ?: false
```

**Comment by marandaneto** [comment]
> throw new Error(`Cache upload failed because file read failed with ${error.message}`);

**Comment by cortinico** [comment]
> throw new Error(`Cache upload failed because file read failed with ${error.message}`);

Those are false negatives from the caching action. I've retriggered them and they're green now.
I'll keep an eye on those if they happen again.

**Comment by cerisier** [comment]
@marandaneto @cortinico Can I be of any help to get this merged :)

I'm running into issues because of current version of the plugin executing the mapping upload work as part of the configuration closure, effectively trying to upload mappings at every single builds for every single variants.

I was about to create a PR to fix just that but stumbled upon this epic PR which effectively solves this as well :)

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+object BuildPluginsVersion {
+    val AGP = System.getenv("VERSION_AGP") ?: "4.1.0"
```
Comment (marandaneto) [inline buildSrc/src/main/java/Dependencies.kt:2]:
```suggestion
    val AGP = System.getenv("VERSION_AGP") ?: "4.1.3"
```

File: buildSrc/src/main/java/Dependencies.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    const val KOTLIN = "1.4.21"
```
Comment (marandaneto) [inline buildSrc/src/main/java/Dependencies.kt:3]:
```suggestion
    const val KOTLIN = "1.4.32"
```

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,5 @@
-#Thu Oct 15 09:46:23 CEST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
```
Comment (marandaneto) [inline gradle/wrapper/gradle-wrapper.properties:3]: should we try 7.0.0?

File: .github/workflows/test-publish-dry-run.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        name: ["stable"]
+        include:
+          - agp: "4.0.0"
+            gradle: "6.1.1"
+            experimental: false
+            name: AGP-4.0.0
+          - agp: "4.1.2"
+            gradle: "6.5"
+            experimental: false
+            name: AGP-4.1.2
```
Comment (marandaneto) [inline .github/workflows/test-publish-dry-run.yaml:30]: lets add an entry for gradle 7.0.0

**Comment by marandaneto** [comment]
> @marandaneto @cortinico Can I be of any help to get this merged :)
> 
> Right now, the plugin eagerly creates tasks immediatly for all variants, I was about to create a PR to fix just that but stumbled upon this epic PR which effectively solves this as well :)

hey @cerisier sorry late reply, we're about to release an alpha version by the end of the next week :)
it'd be awesome to get some validation, I'll keep you posted.

File: plugin-build/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,101 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    kotlin("jvm") version BuildPluginsVersion.KOTLIN
+    id("java-gradle-plugin")
+    id("maven-publish")
```
Comment (marandaneto) [inline plugin-build/build.gradle.kts:6]: what's about using https://github.com/vanniktech/gradle-maven-publish-plugin
the benefit is, it handles sources, javadoc, signing and the usage of `closeAndReleaseRepository` so we don't have to click buttons

File: plugin-build/gradle.properties [hunk 7]
```diff
@@ -6,4 +6,19 @@ version = 1.7.37-SNAPSHOT
...
+# publication pom properties
+POM_NAME=Sentry Android Gradle Plugin
+POM_DESCRIPTION=Sentry Android Gradle Plugin
+POM_URL=https://github.com/getsentry/sentry-android-gradle-plugin
+POM_SCM_URL=https://github.com/getsentry/sentry-android-gradle-plugin
+POM_SCM_CONNECTION=scm:git:git://github.com/getsentry/sentry-android-gradle-plugin.git
+POM_SCM_DEV_CONNECTION=scm:git:ssh://git@github.com/getsentry/sentry-android-gradle-plugin.git
+
+POM_LICENCE_NAME=MIT
+POM_LICENCE_URL=http://www.opensource.org/licenses/mit-license.php
```
Comment (bruno-garcia) [inline plugin-build/gradle.properties:20]: What about changing the default branch to `main` when swap to the new code base?
Good chance to do that.

Meanwhile: 
```suggestion
POM_LICENCE_URL=https://github.com/getsentry/sentry-android-gradle-plugin/blob/master/LICENSE
```

Comment (cortinico) [reply plugin-build/gradle.properties]: Yup absolutely 👌

**Comment by marandaneto** [comment]
if it's green, we can merge to master :)

**Pull request merged**